packages/recommend-core/src/utils/mapByScoreToRecommendations.ts

```diff
@@ -1,21 +1,21 @@
-import { ProductRecord } from '../types';
+/* eslint-disable dot-notation */
 
 import { sortBy } from './sortBy';
 
-type MapToRecommendations<TObject> = {
-  hits: Array<ProductRecord<TObject>>;
+type MapToRecommendations<THit> = {
+  hits: THit[];
   maxRecommendations?: number;
 };
 
-export function mapByScoreToRecommendations<TObject>({
+export function mapByScoreToRecommendations<THit>({
   hits,
   maxRecommendations,
-}: MapToRecommendations<TObject>): Array<ProductRecord<TObject>> {
+}: MapToRecommendations<THit>): THit[] {
   // Since recommendations from multiple indices are returned, we
   // need to sort them descending based on their score.
-  return sortBy<ProductRecord<TObject>>((a, b) => {
-    const scoreA = a._score || 0;
-    const scoreB = b._score || 0;
+  return sortBy<THit>((a, b) => {
+    const scoreA = a['_score'] ? a['_score'] : 0;
+    const scoreB = b['_score'] ? b['_score'] : 0;
```

Comment:
Isn't `_score` always returned by the Recommend API?

If not, perhaps we can change the generic to something like `THit extends { _score?: number }`.